File: chapter4/java/designpattern/src/main/java/com/designpattern/SpringWater.java
package com.designpattern;

/**
 * SpringWater product information
 *
 * @Author Bridget Wu
 */
public class SpringWater implements Product {

    private int id = 1;

    private String name = "Spring Water";

    private int price;

    public SpringWater(int price) {
        this.price = price;
    }

    @Override
    public int getId() {
        return this.id;
    }

    @Override
    public int getPrice() {
        return this.price;
    }

    @Override
    public String getName() {
        return this.name;
    }
}

File: chapter4/typescript/SpringWater.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpringWater = void 0;
/**
 * SpringWater product information
 *
 * @Author Bridget Wu
 */
class SpringWater {
    constructor(price) {
        this.id = 1;
        this.name = "Spring Water";
        this.price = 0;
        this.price = price;
    }
    getId() {
        return this.id;
    }
    getPrice() {
        return this.price;
    }
    getName() {
        return this.name;
    }
}
exports.SpringWater = SpringWater;
